lib/cc/analyzer/formatters/html_formatter.rb

```diff
@@ -0,0 +1,104 @@
+require "redcarpet"
+require "active_support/number_helper"
+
+module CC
+  module Analyzer
+    module Formatters
+      class HTMLFormatter < Formatter
+        def started
+          header = <<-EOD
+          <html>
+            <head>
+            <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
+            <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap-theme.min.css" integrity="sha384-fLW2N01lMqjakBkx3l/M9EahuwpSfeNvV63J5ezn3uZzapT0u7EYsXMjQV+0En5r" crossorigin="anonymous">
+            <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.2/jquery.min.js"></script>
+            <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script>
+            </head>
+            <body>
+            <div class="container">
+            <h2>Code Climate Analysis</h2>
+          EOD
+          puts header
+        end
+
+        def write(data)
+          json = JSON.parse(data)
+          json["engine_name"] = current_engine.name
+
+          case json["type"].downcase
+          when "issue"
+            issues << json
+          when "warning"
+            warnings << json
+          else
+            raise "Invalid type found: #{json["type"]}"
+          end
+        end
+
+        def finished
+          puts "<h3>#{pluralize(issues.size, "issue")} found in #{issues_by_path.size} files.</h3>"
```

Comment:
All these `puts` make me sad. How hard would it be to pull something like ERB into this to handle these?